### PR TITLE
Feat/13 14 15 updating api schemas and adding basic route descriptions

### LINKED ISSUE #13: Use CamelCase
## Detailed Description
make sure CamelCase for pydantic classes

## Context
Good to keep things consistent

## Possible Implementation
- Change `PV_Actual_Value` to `PVActualValue` and others

### PR DESCRIPTION
**Pull request by rachel-labri-tipton**
# Pull Request

### Description

- Adds basic descriptions to the PV Site API routes
- Changes the pydantic model names to CamelCase
- Updates pydantic models to better align with the database architecture 

Questions: 

- I've put `.pyc` files in the `.gitignore` file and only staged changes in the `main.py` file and the `pydantic_models.py` file, and yet, the `pyc` files got pushed. :/ How can I avoid this? Will delete when I make changes in the code tomorrow.

- These current changes don't account for the `datetime_interval_uuid` that will be part of the database tables. do I need to include `datetime_interval_uuid` in the API routes and data models? 

Please let me know what other tidying needs to be done and/or if the route descriptions need to be more developed. The top of the Swagger UI still says FastAPI, and I didn't figure out yet how to change that. Probably just need to grab the code from the other Nowcasting API. 

Fixes #13, #14, #15

## How Has This Been Tested?

I ran the code locally with the FastAPI Swagger interface to view the API updates.

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -5,39 +5,36 @@
...
-    Multiple_PV_Actual,
-    PV_Sites,
+    PVSiteMetadata,
+    MultiplePVActual,
+    PVSites,
     PVSiteAPIStatus,
-    Site_Forecast_Values,
-    PV_Actual_Value
+    SiteForecastValues,
+    PVActualValue
```
Comment (peterdudfield) [inline src/main.py:13]: nice

File: src/main.py [hunk 2]
```diff
@@ -113,49 +133,64 @@ async def get_pv_actual(site_uuid: str):
...
-@app.get("/sites/pv_forecast/metadata/{forecast_metadata_uuid}", response_model=Forecast_Metadata)
-async def get_forecast_metadata(forecast_metadata_uuid: str):
-
-    fake_forecast_metadata = Forecast_Metadata(
-        forecast_metadata_uuid=forecast_metadata_uuid,
-        site_uuid=fake_site_uuid,
-        forecast_creation_datetime=datetime.now(timezone.utc),
-        forecast_version="0.0.1",
-    )
+# @app.get("/sites/pv_forecast/metadata/{forecast_metadata_uuid}", response_model=ForecastMetadata)
```
Comment (peterdudfield) [inline src/main.py:172]: delete, always in the git history

File: src/pydantic_models.py [hunk 3]
```diff
@@ -64,25 +70,26 @@ class Site_Forecast_Values(BaseModel):
...
+    forecast_values: List[SiteForecastValues] = Field(
          ..., description="List of target times and generation"
      )
 
 
 # client gets information about the forecast: which site, which forecast version, when it was made
 # parameters: site_uuid
-class Forecast_Metadata(BaseModel):
-    """Forecast with site information"""
+# class ForecastMetadata(BaseModel):
```
Comment (peterdudfield) [inline src/pydantic_models.py:83]: remove

**Review by peterdudfield — APPROVED** [review]
a few mall changes, but looks really good. Don;t worry about he datetime_interval stuff, lets keep it like this

**Comment by peterdudfield** [comment]
> a few mall changes, but looks really good. Don;t worry about he datetime_interval stuff, lets keep it like this

oh, we should make sure tests and lint are passing, before merging

**Comment by peterdudfield** [comment]
for the `.pyc` files you have to activity delete them (or git, stop following those files). This allows file endings to be put in the gitignore, but still we could commit some e.g ignore all images, but commit this one only

**Comment by rachel-labri-tipton** [comment]
thanks, @peterdudfield. i'll make the changes. for like the 500th time, would you be able to tell me what the command is that I need to run locally to check for linting errors?  `pip3 flake8 run`? i promise to write it down this  time. actually, i'll just put it in the readme.

**Comment by peterdudfield** [comment]
> thanks, @peterdudfield. i'll make the changes. for like the 500th time, would you be able to tell me what the command is that I need to run locally to check for linting errors? `pip3 flake8 run`? i promise to write it down this time. actually, i'll just put it in the readme.

The easiest way is to go and look at the failed CI and copy it from there
Screenshot 2023-01-12 at 11 40 54

**Comment by rachel-labri-tipton** [comment]
hey @peterdudfield, Just following up on slack comments here. These are the two error messages from the tests that are failing that I couldn't figure out how to resolve:  Screenshot 2023-01-12 at 15 35 30
The full error message for the second failed test is: TypeError: Object of type datetime is not JSON serializable. I didn't quite know what to do with this even after a bit of googling. Maybe you've seen it before?

**Comment by peterdudfield** [comment]
> hey @peterdudfield, Just following up on slack comments here. These are the two error messages from the tests that are failing that I couldn't figure out how to resolve: Screenshot 2023-01-12 at 15 35 30 The full error message for the second failed test is: TypeError: Object of type datetime is not JSON serializable. I didn't quite know what to do with this even after a bit of googling. Maybe you've seen it before?

now sorted in the pushed ive made

**Comment by peterdudfield** [comment]
nice one!

**Pull request merged**